aspnetcore/performance/performance-best-practices.md

```diff
@@ -0,0 +1,132 @@
+
+## Keep common code paths fast
+
+You want all of your code to be fast, but frequently called code paths are the most critical to optimize:
+
+* Middleware components in the app's request processing pipeline, especially middleware run early in the pipeline. These components have a large impact on performance.
+* Code that is executed for every request or multiple times per request. For example, custom logging, authorization handlers, or initialization of transient services.
+
+Recommendations:
+
+* **Do not** use custom middleware components with long-running tasks.
+* **Do** use performance profiling tools (like [Visual Studio Diagnostic Tools](/visualstudio/profiling/profiling-feature-tour) or [PerfView](https://github.com/Microsoft/perfview)) to identify [hot code paths](#hot).
+
+## Complete long-running Tasks outside of HTTP requests
+
+Most requests to an ASP.NET Core app can be handled by a controller or page model calling necessary services and returning an HTTP response. For some requests which involve long-running tasks, it's better to make the entire request-response process asynchronous.
+
+Recommendations:
+
+* **Do not** wait for long-running tasks to complete as part of ordinary HTTP request processing.
+* **Do** consider handling long-running requests with [background services](/aspnet/core/fundamentals/host/hosted-services) or out of process with an [Azure Function](/azure/azure-functions/). Completing work out-of-process is especially beneficial for CPU-intensive tasks.
+* **Do** use real-time communication options like [SignalR](xref:signalr/introduction) to communicate with clients asynchronously.
+
+## Minify client assets
+
+ASP.NET Core apps with complex front-ends frequently serve many JavaScript, CSS, or image files. Performance of initial load requests can be improved by:
+
+* Bundling, which combines multiple files into one.
+* Minifying, which reduces the size of files by.
+
+Recommendations:
+
+* **Do** use ASP.NET Core's [built-in support](xref:client-side/bundling-and-minification) for bundling and minifying client assets.
+* **Do** consider other third-party tools like [Gulp](uid:client-side/bundling-and-minification#consume-bundleconfigjson-from-gulp) or [Webpack](https://webpack.js.org/) for more complex client asset management.
+
+## Use the latest ASP.NET Core release
+
+Each new release of ASP.NET includes performance improvements. Optimizations in .NET Core and ASP.NET Core mean that newer versions will outperform older versions. For example, .NET Core 2.1 added support for compiled regular expressions and benefitted from [`Span<T>`](https://msdn.microsoft.com/magazine/mt814808.aspx). ASP.NET Core 2.2 added support for HTTP/2. If performance is a priority, consider upgrading to the most current version of ASP.NET Core.
+
+<!-- TODO review link and taking advantage of new [performance features](#TBD)
```

Comment:
I think there's value in having section of our docs that groups performance features (things like caching, response compression, etc.). This could just link to that section.